### ISSUE TITLE
Build failing in production (DigitalOcean) due to Time crate dependency.

### ISSUE THREAD

**Issue by parK-dev** [actix/actix-web#2937]
```
Updating crates.io index
[2022-11-27 14:45:58] │ error: failed to select a version for the requirement `time = "^0.3"`
[2022-11-27 14:45:58] │ candidate versions found which didn't match: 0.3.15, 0.3.14, 0.3.13, ...
[2022-11-27 14:45:58] │ location searched: crates.io index
[2022-11-27 14:45:58] │ required by package `actix-web v4.2.1`
[2022-11-27 14:45:58] │     ... which satisfies dependency `actix-web = "^4"` (locked to 4.2.1) of package `newsletter-api-rust v0.0.1 (/app)`
[2022-11-27 14:45:58] │ thread 'main' panicked at 'Exited with status code: 101', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/cargo-chef-0.1.35/src/recipe.rs:145:27
[2022-11-27 14:45:58] │ note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[2022-11-27 14:45:58] │ error building image: error building stage: failed to execute command: waiting for process to exit: exit status 101
[2022-11-27 14:45:58] │ 
[2022-11-27 14:45:58] │ command exited with code 101
```
- Actix Web Version: 4

**Comment by JohnTitor** [comment]
Could you provide your dependencies list and reproducible steps? actix-web only specifies `time = "0.3"` which means all the 0.3.x is allowed and by itself, it shouldn't cause the problem.

**Comment by parK-dev** [comment]
```
[dependencies]
actix-web = "4"
serde = { version= "1", features = ["derive"] }
tokio = { version = "1", features = ["macros", "rt-multi-thread"] }
config = "0.13.2"
chrono = { version = "0.4.23", default-features = false, features = ["clock"] }
uuid = { version = "1.2.2", features = ["v4"] }
tracing = { version = "0.1.37", features = ["log"] }
tracing-subscriber = { version = "0.3.16", features = ["registry", "env-filter"] }
tracing-bunyan-formatter = "0.3.4"
tracing-log = "0.1.3"
secrecy = { version = "0.8.0", features = ["serde"] }
tracing-actix-web = "0.6.2"
serde-aux = "4.1.2"
time = "0.3.15"

[dependencies.sqlx]
version = "0.6"
default-features = false
features = [
    "runtime-tokio-rustls", 
    "macros", 
    "postgres", 
    "uuid", 
    "chrono", 
    "migrate",
    "offline"
]

[dev-dependencies]
once_cell = "1.16.0"
reqwest = "0.11.13"

```

**Comment by parK-dev** [comment]
It builds and runs fine when I use cargo build --release locally. Somehow the issue only happens when building on OceanDigital app.

**Comment by robjtede** [comment]
Which Rust version is being used on DO?

**Comment by rootBiao** [comment]
now,I happen this issue too.

**Comment by JohnTitor** [comment]
I've come up with two possibilities.

1.  `cargo-chef` does something wrong and cached deps cause the issue.
	- If it's managed by you, try to remove it. 
3. a specific cargo version cannot resolve deps correctly.
  	- providing a cargo version would be helpful to investigate.

A reproducible way would also be helpful!

**Comment by ShadowStrikerJ** [comment]
I was able to reproduce this issue using version `1.59.0` of cargo and rust, but the issue was resolved upgrading to `1.66.0`
